walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -31,55 +46,106 @@
+    // -1 for mandatory attendees for 1 and 3, because it's the end of the day/A goes to the meeting
+    HashMap<Integer, AvailabilityChange> timePointsOfAvailabilityChange = new HashMap<>();
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
 
-    long numberOfAttendeesRequired = attendees.size();
+    long numOfMandAttendeesRequired = mandatoryAttendees.size();
+    long numOfOptAttendeesRequired = optionalAttendees.size();
+    long numOfAttendeesRequired = numOfMandAttendeesRequired + numOfOptAttendeesRequired;
     long requiredDuration = request.getDuration();
 
     // add start and end times of the day
-    timePointsOfAvailabilityChange.put(0, numberOfAttendeesRequired);
-    timePointsOfAvailabilityChange.put(TimeRange.WHOLE_DAY.end(), -numberOfAttendeesRequired);
+    timePointsOfAvailabilityChange.put(0,
+            new AvailabilityChange(numOfMandAttendeesRequired,
+                    numOfOptAttendeesRequired));
+    timePointsOfAvailabilityChange.put(TimeRange.WHOLE_DAY.end(),
+            new AvailabilityChange(-numOfMandAttendeesRequired,
+                    -numOfOptAttendeesRequired));
 
     // add all time point, where the number of free attendees is changing
-    for (Event event : events) {
+    for (Event existingEvent : events) {
       // the event can have attendees that are not present in the request,
       // so we shouldn't count them
-      Collection<String> attendeesOfExistingEvent = new ArrayList<>(event.getAttendees());
-      attendeesOfExistingEvent.retainAll(attendees);
-
-      // if intersection isn't empty
-      if (!attendeesOfExistingEvent.isEmpty()) {
-        int numAttendeesOfExistingEvent = attendeesOfExistingEvent.size();
-        int startTime = event.getWhen().start();
-        int endTime = event.getWhen().end();
-
-        timePointsOfAvailabilityChange.put(startTime,
-                timePointsOfAvailabilityChange.getOrDefault(startTime, 0L) - numAttendeesOfExistingEvent);
-        timePointsOfAvailabilityChange.put(endTime,
-                timePointsOfAvailabilityChange.getOrDefault(endTime, 0L) + numAttendeesOfExistingEvent);
+      Collection<String> mandAttendeesOfExistingEvent = new ArrayList<>(existingEvent.getAttendees());
+      mandAttendeesOfExistingEvent.retainAll(mandatoryAttendees);
+      Collection<String> optAttendeesOfExistingEvent = new ArrayList<>(existingEvent.getAttendees());
+      optAttendeesOfExistingEvent.retainAll(optionalAttendees);
+
+      // if intersection isn't empty, then it's a time moment, where the available
+      // number of attendees changes, in other case, we skip this event
+      if (mandAttendeesOfExistingEvent.isEmpty() && optAttendeesOfExistingEvent.isEmpty()) {
+        continue;
       }
+
+      AvailabilityChange timePointChanges;
+      long changeOfMandAttendees, changeOfOptAttendees;
```

Comment:
Google's style guide recommends only one variable definition per line. Split this in two lines